scripts/check_change_log/main.py

```diff
@@ -0,0 +1,121 @@
+# coding=utf-8
+import glob
+import os
+import subprocess as sp
+import time
+from pypi_tools.pypi import PyPIClient
+from pathlib import Path
+
+error_packages_info = {}
+
+def find_report_name(result):
+    signal_api_pattern = 'written to'
+    multi_api_pattern = 'merged_report'
+    print("result: ", result)
+    for line in result:
+        idx = line.find(signal_api_pattern)
+        idx1 = line.find(multi_api_pattern)
+        if idx > 0 and idx1 > 0:
+            return "_/" + line[idx + len(signal_api_pattern):].replace(" ", "")
+
+    for line in result:
+        idx = line.find(signal_api_pattern)
+        if idx > 0:
+            return "_/" + line[idx + len(signal_api_pattern):].replace(" ", "")
+
+    return ''
+
+def create_folder(name):
+    if not os.path.exists(name):
+        os.mkdir(name)
+        print("create folder successfully")
+    else:
+        print("folder has been created")
+
+def write_txt(foldor, text_name, text, last_version, older_version):
+    path = foldor + f"\{text_name}" + " " + last_version + "-" + older_version + r".txt"
+    with open(file=path, mode="w", encoding="utf-8") as file:
+
+        file.write(text)
+    print("change_log.txt create successful")
+
+def create_code_report(cmd, service_name):
+    info = sp.Popen(cmd,
+                    stderr=sp.STDOUT,
+                    stdout=sp.PIPE,
+                    universal_newlines=True,
+                    cwd=None,
+                    shell=False,
+                    env=None,
+                    encoding="utf-8",
+                    )
+    output_buffer = []
+    for line in info.stdout:
+        output_buffer.append(line.rstrip())
+    info.wait()
+    info_output = "\n".join(output_buffer)
+    if info.returncode:
+        err_info = '\n'.join(output_buffer[-min(len(output_buffer), 7):0])
+        error_packages_info[service_name] = err_info
+    return info_output.split('\n')
+
+if __name__ == '__main__':
+    # get sdk path
+    env = Path.cwd()
+    docker_path = env.parent.parent
+
+    # create docker env in sdk path
+    sp.call(fr"docker create -it --rm -h Change_log --name Change_log -v {docker_path}:/_ l601306339/autorest")
+    sp.call("docker start Change_log")
+
+    # install azure tools
+    sp.call(f'docker exec -it Change_log /bin/bash -c  "python _/scripts/dev_setup.py -p azure-core"  ')
+
+    # get all azure-mgmt-package paths
+    in_files = glob.glob(str(Path(f'{docker_path}/sdk/*/azure-mgmt-*')))
+    for i in in_files[0:1]:
+        path = Path(i)
+        service_name = path.parts[-1]
+        print(service_name)
+
+        # get package version in pypi
+        client = PyPIClient()
+        versions = [str(v) for v in client.get_ordered_versions(f"{service_name}")]
```

Comment:
```suggestion
        versions = [str(v) for v in client.get_ordered_versions(f"{service_name}")]
```
why f"{service_name}" ?